pallets/collator-selection/src/benchmarking.rs

```diff
@@ -50,12 +51,23 @@ fn assert_last_event<T: Config>(generic_event: <T as Config>::Event) {
 	assert_eq!(event, &system_event);
 }
 
-fn register_candidates<T: Config>(count: u32) {
+fn register_candidates<T: Config>(count: u32, removals: u32) {
 	let candidates = (0..count).map(|c| account("candidate", c, SEED)).collect::<Vec<_>>();
 	assert!(<CandidacyBond<T>>::get() > 0u32.into(), "Bond cannot be zero!");
+	let mut i = 0;
 	for who in candidates {
 		T::Currency::make_free_balance_be(&who, <CandidacyBond<T>>::get() * 2u32.into());
-		<CollatorSelection<T>>::register_as_candidate(RawOrigin::Signed(who).into()).unwrap();
+
+		let last_block = if count - i > removals {10u32.into()} else {0u32.into()};
```

Comment:
```suggestion
		let last_block = if count - i > removals { 10u32.into() } else { 0u32.into() };
```

Reply:
Also, the relationship between this 10 and the `removal` can probably be enhanced. Looking at your benchmark, you want to have some candidates kicked at each new session. I think a nicer way of doing this is:

- leave this function as-is, make it always set `last_block` to `now`.
- in the benchmark that is interested in this, manually set `last_block` of `r` of these folks to `now - (boot + 1)` (as opposed to the magic 0 and 10).